EIPS/eip-3220.md

```diff
@@ -0,0 +1,121 @@
+
+## Abstract
+
+Crosschain operations include transfer of assets, data, message, events, and commands from one blockchain to another. In order to improve interoperability of crosschain operations, there is a need to uniquely identify source blockchain, target blockchains, and any relay chains that help facilite blockchain information transfer.  The identifier should follow a standard format so that all platforms and decentralize applications can use this identifier to represent a blockchain and also some information in the identifier to get some characteristis of the blockchain.  We name this blockchain identifier as crosschain-id and expect Ethereum community to use crosschain-id to represent any public or private blockchain.  The crosschain-id is a 32 byte hex string and with some bytes extracted from blockchain hash and some manually defined to characterize a blockchain.  We also propose a registration and lookup service to retrieve blockchain metadata from the crosschain-id.
+
+## Motivation
+
+EIP-155 was proposed by Vitalik to provide a unique identifier to a blockchain to provide simple relay attack protection.  This specification defines an integer for Chainid for a blockchain and sign the chainid into a transaction data and hence present attackers to send same transaction to different blockchains. This specification will require blockchains to define a chainid and register the chainid in a public repository.
+
+The challenge of using an integer for chainid is that it is not broad enough to cover all blockchains and it does not prevent different blockchains using the same chainid.  Also, it does not address the issue for two forked blockchains having the same chainid.
+
+Hence there is need for a more robust blockchain identifier that will overcome these drawbacks, especially for crosschain operations where multiple chains are involved. A blockchain identifier (crosschain id) should be unique and satisfy the following requirements:
+
+should provide identification, description, and discovery of blockchains.
+should provide unique identification of each blockchain in the crosschain service ecosystem.
+should provide descriptions for a blockchain identities such as chainId, name, type, consensus scheme etc.
+should provide discovery mechanism for supported blockchains and also for new blockchains in the ecosystem.
+should provide a mechanism for a joining blockchain to register to the ecosystem.
+should provide a mechanism for a blockchain to edit properties or unregister from the crosschain ecosystem.
+should provide a mechanism to get some critical information of the blockchain
+should provide a mechanism to differentiate an original blockchain and a forked blockchain
+should provide a mechanism to verify a chainid without external registration service
```

Comment:
Recommend a bulleted list.  In markdown this will render as a paragraph, without proper capitalization.

Reply:
Modified